client/app/bundles/course/assessment/submission/components/ReadOnlyEditor/index.jsx

```diff
@@ -5,7 +5,7 @@ import PropTypes from 'prop-types';
 
 import { annotationShape, fileShape } from '../../propTypes';
 import translations from '../../translations';
-import ProgrammingFileDownloadLink from '../answers/Programming/ProgrammingFileDownloadLink';
+import ProgrammingFilenameWithDownload from '../answers/Programming/ProgrammingFilenameWithDownload';
```

Comment:
Any reason why only ReadOnlyEditor is changed?
What about when file is too big? Or when submission is in attempting state, using editor mode?

If there is no reason for `ProgrammingFileDownloadLink` to exist, we can replace all `ProgrammingFileDownloadLink`s to `ProgrammingFilenameWithDownload` instead.

Reply:
You're not resolving the user's ask:
> Just took a look at the student submissions, the file contents are all pasted in the wrong locations as a result.  

You can check `/courses/2698/assessments/66548/submissions/2051201/edit`, which has is a programming question created through Evaluation package > `Manually create/edit offline and upload` > Upload package. 

In this form, then `multiple_file_submission` flag is always set to `false`, students cannot upload new files, but it's entirely possible for instructor to have have multiple template files in the uploaded package.

The problem of students not knowing which file is which still exists. Either entirely replace `ProgrammingFileDownloadLink` to `ProgrammingFilenameWithDownload` or figure out a way to detect if there are multiple files in the answer, e.g. submission json `answers.[idx].fields.file_attributes > 1` and render the chips.